src/async_wrap.h

```diff
@@ -217,6 +219,9 @@ class AsyncWrap : public BaseObject {
   // Because the values may be Reset(), cannot be made const.
   double async_id_ = kInvalidAsyncId;
   double trigger_async_id_;
+  // Flag to track state of promise
+  // All promises are initialised as unsettled
+  bool has_settled_ = false;
```

Comment:
Since all of this is specific to Promises, I’d put it on the `PromiseWrap` class instead